### PR TITLE
autogtp: make tuning before starting threads

### PR DESCRIPTION
**Pull request by marcocalignano**
I guess this you will like more. The tuning process is also verbose.

### CODE REVIEW

File: autogtp/Management.cpp [hunk 1]
```diff
@@ -57,8 +57,36 @@ Management::Management(const int gpus,
...
+        tuneProcess.waitForReadyRead(1000);
+        QTextStream(stdout) << tuneProcess.readAllStandardError();
+    }
+    tuneProcess.waitForFinished(-1);
+}
+
 void Management::giveAssignments() {
     sendAllGames();
+
+    //Make the OpenCl tunig before starting the threads
```
Comment (gcp) [inline autogtp/Management.cpp]: nit: typo "tuning"

File: autogtp/Management.cpp [hunk 2]
```diff
@@ -57,8 +57,36 @@ Management::Management(const int gpus,
...
+    QTextStream(stdout) << "Start tuning process please wait ..." << endl;
```
Comment (gcp) [inline autogtp/Management.cpp]: "Starting tuning process, please wait..."

File: autogtp/Management.cpp [hunk 3]
```diff
@@ -57,8 +57,36 @@ Management::Management(const int gpus,
     m_fallBack(Order::Error) {
 }
 
+void Management::runTuningProcess(const QString &tuneCmdLine) {
+    QTextStream(stdout) << tuneCmdLine << endl;
+    QProcess tuneProcess;
+    tuneProcess.start(tuneCmdLine);
```
Comment (gcp) [inline autogtp/Management.cpp]: If the user is running a CPU only client, it will reject --tune-only, so you need to handle this case and just bail out.

**Comment by gcp** [comment]
Thanks!

File: autogtp/Management.cpp (hunk 3 above)
Comment (marcocalignano) [reply autogtp/Management.cpp]: But the process will run, say something like "no OpenCl capable Device found" and exit. The code should not have any problem with this.

**Pull request merged**